content/guides/recipes/screen-resizing.md

```diff
@@ -0,0 +1,186 @@
+
+```lua
+function screen.resize(w, h)
+    local scale = math.min(w / screen.width, h / screen.height)
+    local x = (w - screen.width * scale) / 2
+    local y = (h - screen.height * scale) / 2
+
+    screen.transform:setTransformation(x, y, 0, scale, scale)
+end
+
+return screen
+```
+
+Back in main.lua we need to call our new functions.
+
+```lua
+local screen = require "screen"
+
+screen.init(960, 540)
+
+function love.draw()
+    screen.capture()
+    -- Draw your game
+    screen.draw()
+end
+
+function love.resize(w, h)
+    -- Upon resizing the window, pass the new width and height.
+    screen.resize(w, h)
+end
+```
+
+Now our game will be resized along with our window. There is a slight problem, however. When we draw something based on the position of our mouse, you will see that it doesn't match your mouse's position.
+
+```lua
+function love.draw()
+    screen.capture()
+
+    local x, y = love.mouse.getPosition()
+    love.graphics.circle("fill", x, y, 10)
+
+    screen.draw()
+end
+```
+
+This is because originally it is drawn on our mouse position, but then as the window resizes and shifts, so does the circle we drew. To fix this, we can add a function to get our actual mouse position, or rather what would be the mouse position after the scaling and translating is applied.
+
+```lua
+function screen.getMousePosition()
+    local x, y = love.mouse.getPosition()
+    return screen.transform:inverseTransformPoint(x, y)
+end
+```
+
+The function `inverseTransformPoint` inverses the transformation we apply. The `x` and `y` it gives us are where we need to draw our circle if we want it to appear at the original `x` and `y` after the transformation has been applied.
+
+```lua
+function love.draw()
+    screen.capture()
+
+    local x, y = screen.getMousePosition()
+    love.graphics.circle("fill", x, y, 10)
+
+    screen.draw()
+end
+```
+
+Now our circle draws on the correct position!
+
+## Libraries
+
+For more advanced options, check out the libraries [push](https://github.com/Ulydev/push) and [shove](https://github.com/Oval-Tutu/shove).
+
+## Full code
+
+```lua
+local screen = {}
+
+function screen.init(width, height)
+    screen.width = width
+    screen.height = height
+    screen.canvas = love.graphics.newCanvas(width, height)
+    screen.transform = love.math.newTransform()
+end
+
+function screen.capture()
+    -- Push the current state (including the main canvas) onto the stack.
+    love.graphics.push("all")
+    -- Set our canvas. From now on everything will be drawn on our canvas.
+    love.graphics.setCanvas(screen.canvas)
+    love.graphics.clear()
+end
+
+function screen.draw()
+    -- Pop the state, back to the main canvas.
+    love.graphics.pop()
+    -- Draw our canvas using the transform.
+    love.graphics.draw(screen.canvas, screen.transform)
+end
+
+function screen.resize(w, h)
+    local scale = math.min(w / screen.width, h / screen.height)
```

Comment:
worth mentioning integer only scaling for perfect pixels?
math.flooring the scale here